src/App.tsx

```diff
@@ -1,29 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const items = getNumbers(1, 42).map(n => `Item ${n}`);
+  const total = items.length;
+
+  const startIndex = (currentPage - 1) * perPage + 1;
+  const endIndex = Math.min(currentPage * perPage, total);
+  const visibleItems = items.slice(startIndex - 1, endIndex);
```

Comment:
The calculation for `startIndex` and `endIndex` seems off. `startIndex` should be `(currentPage - 1) * perPage` without the `+ 1`, and `endIndex` should be `currentPage * perPage`. This will ensure correct slicing of the `items` array.